### PR TITLE
adding __eq__ and __str__ to Dimension to make unit testing easier

### PR DESCRIPTION
**Pull request by kdsudac**
This change is [Reviewable](https://reviewable.io/reviews/google/openhtf/713)

### CODE REVIEW

File: openhtf/core/measurements.py [hunk 1]
```diff
@@ -348,6 +348,12 @@ def __init__(self, description='', unit=units.NO_DIMENSION):
     self.description = description
     self.unit = unit
 
+  def __eq__(self, other):
```
Comment (Kenadia) [inline openhtf/core/measurements.py]: To handle Python 2, when implementing \_\_eq__, implement \_\_ne__ as well

```python
def __ne__(self, other):
  return not self == other
```

discussion: https://stackoverflow.com/questions/4352244/python-should-i-implement-ne-operator-based-on-eq

File: openhtf/core/measurements.py [hunk 2]
```diff
@@ -348,6 +348,12 @@ def __init__(self, description='', unit=units.NO_DIMENSION):
...
+    return (self.description == other.description and self.unit == other.unit)
+
+  def __str__(self):
+    return str(self._asdict())
```
Comment (Kenadia) [inline openhtf/core/measurements.py]: Let's use the pattern commonly used in OpenHTF of including the class name.

Also I think this should be \_\_repr__ not \_\_str__. And \_\_str__ will then default to \_\_repr__.

```python
def __repr__(self):
  return '<%s: %s>' % (type(self).__name__, self._asdict())
```

File: openhtf/core/measurements.py (hunk 1 above)
Comment (Kenadia) [reply openhtf/core/measurements.py]: (meant to “request changes)

File: openhtf/core/measurements.py (hunk 2 above)
Comment (Kenadia) [reply openhtf/core/measurements.py]: (meant to “request changes)

File: openhtf/core/measurements.py (hunk 1 above)
Comment (kdsudac) [reply openhtf/core/measurements.py]: Done.

File: openhtf/core/measurements.py (hunk 2 above)
Comment (kdsudac) [reply openhtf/core/measurements.py]: Done.

**Comment by kdsudac** [comment]
Thanks for the review and suggestions.

**Pull request merged**